client/relay-chain-interface/src/lib.rs

```diff
@@ -0,0 +1,441 @@
+use std::{sync::Arc, time::Duration};
+
+use cumulus_primitives_core::{
+	relay_chain::{
+		v1::{
+			CommittedCandidateReceipt, OccupiedCoreAssumption, ParachainHost, SessionIndex,
+			ValidatorId,
+		},
+		Block as PBlock, BlockId, Hash as PHash, InboundHrmpMessage,
+	},
+	InboundDownwardMessage, ParaId, PersistedValidationData,
+};
+use parking_lot::RwLock;
+use polkadot_client::{ClientHandle, ExecuteWithClient, FullBackend};
+use polkadot_service::{
+	AuxStore, BabeApi, CollatorPair, Configuration, Handle, NewFull, Role, TaskManager,
+};
+use sc_client_api::{
+	blockchain::BlockStatus, Backend, BlockchainEvents, HeaderBackend, UsageProvider,
+};
+use sc_telemetry::TelemetryWorkerHandle;
+use sp_api::{ApiError, BlockT, ProvideRuntimeApi};
+use sp_consensus::SyncOracle;
+use sp_core::{sp_std::collections::btree_map::BTreeMap, Pair};
+use std::sync::Mutex;
+
+const LOG_TARGET: &str = "relay-chain-interface";
+
+/// Should be used for all interaction with the relay chain in cumulus.
+pub trait RelayChainInterface<Block: BlockT> {
```

Comment:
Why is that generic over the block?

Reply:
And we could here directly require `Send + Sync` as super traits. Then we don't need to add this when we want to use this trait.